crates/registry/src/package.rs

```diff
@@ -30,9 +31,11 @@ impl Package {
         name: &str,
         http_client: &reqwest::Client,
         registry: &str,
+        semaphore: &Semaphore,
     ) -> Result<Self, RegistryError> {
         let url = || format!("{registry}{name}"); // TODO: use reqwest URL directly
         let network_error = |error| NetworkError { error, url: url() };
+        let _permit = semaphore.acquire().await;
```

Comment:
Is there a reason you gave this value a name? If you want to postpone dropping it, I recommend explicitly calling `drop` (i.e. `drop(permit)`) at the end of the scope. If you only want to `.await` the semaphore, please remove `let _permit =`.

Reply:
won't it be dropped immediately if I don't assign it to a value? As far as I understand, the permit is dropped, when it goes out of scope.